content/6_Advanced/Offline_Del.mdx

```diff
@@ -1,85 +1,269 @@
 ---
 id: offline-del
 title: 'Offline Deletion'
-author: Benjamin Qi, Siyong Huang
+author: Benjamin Qi, Siyong Huang, Chongtian Ma
 prerequisites:
   - dsu
 description: 'Erasing from non-amortized insert-only data structures.'
 frequency: 0
 ---
 
-## Offline Deleting from a Data Structure
-
 Using a persistent data structure or rollbacking, you are able to simulate
 deleting from a data structure while only using insertion operations.
 
-<Resources>
-	<Resource
-		source="CP-Algorithms"
-		title="Deleting from a data structure in O(T(n) log n)"
-		url="https://cp-algorithms.com/data_structures/deleting_in_log_n.html"
-		starred
-	>
-		includes code (but no explanation) for dynamic connectivity
-	</Resource>
-</Resources>
+## DSU With Rollback
+
+**DSU with Rollback** is an extension to DSU that saves merges and can undo the previous $k$ merges.
+
+<Problems problems="rollback" />
```

Comment:
wait did u provide impl for this problem?

Reply:
yuh both